### ISSUE TITLE
fig login for headless install: alternative login options

### ISSUE THREAD

**Issue by nick-youngblut** [withfig/fig#2286]
### Sanity checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my feature

### Description:

At for Fig `2.13.1`, the only login option for a headless install is `fig login`, followed by authentication via email. I would be great if one could authenticate by providing credentials directly within `fig login` (e.g., `fig login --username XXX --credentials XXX`). Then, one could use 1password cli, GitHub Secrets, or something similar to automatically login to fig upon the creation of a devcontainer (or similar dev env setup).

**Comment by grant0417** [comment]
This is in the works, the plan is to add an environment variable `FIG_TOKEN` which you can set [here](https://github.com/settings/codespaces) that will auth you by default

**Comment by mvgijssel** [comment]
Yes this would be perfect for our devcontainer developer environments at HackerOne, being able to login with a token!

**Comment by nick-youngblut** [comment]
I see that `fig login` now allows for `--token` (using fig 2.14.2), but I can't figure out how to create the user token. @grant0417 can you please explain how to create the token?

**Comment by grant0417** [comment]
You can create a token by running `fig user token new <token-name>` somewhere where you are already logged in, you will get a token that should look like `figapi_xxxxxxxxx` which you can then use with `fig login --token fig_api_xxxxxxxx` to auth another machine.

Note the `<token-name>` is just a description of the token but is required.

**Comment by grant0417** [comment]
There is now a page you can go to that you can generate tokens with and see all of your tokens, https://app.fig.io/settings/user-tokens
